cluster_management/src/main/java/com/pinterest/rocksplicator/OnlineOfflineStateModelFactory.java

```diff
@@ -48,7 +49,7 @@
     this.adminPort = adminPort;
     this.cluster = cluster;
     this.zkClient = CuratorFrameworkFactory.newClient(zkConnectString,
-        new ExponentialBackoffRetry(1000, 3));
+        new BoundedExponentialBackoffRetry(1000,  10000, MAX_ZK_RETRIES));
```

Comment:
ExponentialBackoffRetry is already bounded to retry up to 3 times. Any reason why we need to bound by time as well with BoundedExponentialBackoffRetry?

Reply:
In order to guarantee the max time. The intermediate intervals are randomized... so we can't guarantee on the upperbound on time. This will help keep that latency low, in worst case.